app/src/org/commcare/android/crypt/EncryptionIO.java

```diff
@@ -0,0 +1,95 @@
+package org.commcare.android.crypt;
+
+import android.util.Log;
+
+import org.commcare.android.javarosa.AndroidLogger;
+import org.javarosa.core.services.Logger;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Methods for dealing with encrypted input/output.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class EncryptionIO {
+    private final static String TAG = EncryptionIO.class.getSimpleName();
+
+    public static OutputStream createFileOutputStream(String filename,
+                                                      SecretKeySpec symetricKey)
+            throws FileNotFoundException {
+        final File path = new File(filename);
+        FileOutputStream fos = new FileOutputStream(path);
+        if (symetricKey == null) {
+            return fos;
+        } else {
+            try {
+                Cipher cipher = Cipher.getInstance("AES");
+                cipher.init(Cipher.ENCRYPT_MODE, symetricKey);
+                return new CipherOutputStream(fos, cipher);
```

Comment:
I thought in the case where you have all the data already lying around you don't need to use buffered streams?
Since this is a util method used in several contexts, not just in when writing to a byte array to a file in `HybridFileBackedSqlStorage`, I'll add the buffered stream. But am I confused in saying that in the case of `HybridFileBackedSqlStorage` it isn't needed?